stacktrace-gps.js

```diff
@@ -243,14 +251,24 @@
                 this._get(fileName).then(function (source) {
                     var sourceMappingURL = _findSourceMappingURL(source);
                     var isDataUrl = sourceMappingURL.substr(0, 5) === 'data:';
+                    var base = fileName.substring(0, fileName.lastIndexOf('/') + 1);
 
                     if (sourceMappingURL[0] !== '/' && !isDataUrl && !(/^https?:\/\/|^\/\//i).test(sourceMappingURL)) {
-                        sourceMappingURL = fileName.substring(0, fileName.lastIndexOf('/') + 1) + sourceMappingURL;
+                        sourceMappingURL = base + sourceMappingURL;
                     }
 
                     this._get(sourceMappingURL).then(function (map) {
                         var lineNumber = stackframe.lineNumber;
                         var columnNumber = stackframe.columnNumber;
+
+                        if (typeof map === 'string') {
+                            map = _parseJson(map.replace(/^\)\]\}'/, ''));
```

Comment:
I used this construct because that's how the source-map module is currently parsing the data (cf [here](https://github.com/mozilla/source-map/blob/master/lib/source-map-consumer.js#L16-L18)). As to why they do this, it's quite probably to give the user a way to secure themselves against [JSON hijacking](http://stackoverflow.com/questions/2669690/why-does-google-prepend-while1-to-their-json-responses?lq=1).